pkg/operator/ceph/controller/spec.go

```diff
@@ -75,21 +75,64 @@ var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
 PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/
+LOG_MAX_SIZE=%s
+
+# size of log after removing any alphabet in the end. eg: for '1G' getting '1'
+LOG_MAX_SIZE_IN_DIGIT=$(echo "${LOG_MAX_SIZE}" | sed 's/.$//')
+# get last char of file size. eg: for '1G' getting 'G'
+GET_LAST_CHAR_OF_LOG_SIZE=$(echo $LOG_MAX_SIZE | rev | cut -c1)
+
+if [ "$GET_LAST_CHAR_OF_LOG_SIZE" == "G" ]; then
```

Comment:
Handling sizing like this in bash is not so easy as to just check the suffix. What if a user inputs `1MM` by mistake? How will this parse? What if `1N` by mistake? `1g`? What if they use `1Mi` since they assume it works like other kubernetes inputs? 

Bash just doesn't have good primitives to handle this well. I can't stress this enough. We absolutely should use Go to parse this, which handles it much better and can give better failures? Then the script should just accept the value in MB? After all, the script is generated by Go code anyway.